### PR TITLE
Add form database

### PR DESCRIPTION
**Pull request by EthanThatOneKid**
Introducing `Form` dataclass and database implementations `InMemoryFormDatabase` (unit testing) and `JSONFormDatabase` (development).

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,4 @@
+CLIENT_ID = "hi"
+CLIENT_SECRET = "hi"
+JWT_SECRET = "hi"
+SERVER_URL = "hello
```
Comment (EthanThatOneKid) [inline .env.example:4]:
```suggestion
SERVER_URL = "hello"

```

File: form/pickle_form_database.py
Comment (EthanThatOneKid) [inline form/pickle_form_database.py:1]: [4:23 PM]Ethan (he/Webmaster): TODO: Refactor Pickle storage into JSON storage.

Comment (EthanThatOneKid) [reply form/pickle_form_database.py:1]: Addressed in <https://github.com/acmcsufoss/form/pull/3/commits/b6eaf9977d0dcea214f3c616f02dc21a41178938>.

**Review by 26samaahmed — APPROVED** [review]
Good Progress!

File: response/pickle_response_database.py
Comment (EthanThatOneKid) [inline response/pickle_response_database.py:1]: TODO(@EthanThatOneKid): Implement JSONResponseDatabase methods.

File: form/json_form_database.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    def delete_form(self, form_id: str) -> None:
+        """Delete a form by ID."""
+        form_map = self._load_forms()
+        if form_id in form_map.data.keys():
+            del form_map.data[form_id]
+            self._save_forms(form_map)
+
+    def _load_forms(self) -> form.form.FormMap:
+        with open(self.file_path, "r") as f:
```
Comment (EthanThatOneKid) [inline form/json_form_database.py:36]: Use a try-except statement to remove the need for a `forms.json.example` file.

Comment (EthanThatOneKid) [reply form/json_form_database.py:36]: Followed up by #7.

File: response/pickle_response_database.py
Comment (EthanThatOneKid) [reply response/pickle_response_database.py:1]: Followed up in #8.

**Pull request merged**